File: lab/javascript/tecnicas/dicionarios_e_setsiplace.js
console.log("OBJETO NORMAL:");
let d1 = {a: 1};
let d2 = {b: 2};
let d3 = {...d1, ...d2}; // spread -> novo objeto
console.log(d1, d2, d3); // {a:1} {b:2} {a:1, b:2}

console.log("OBJETO IN-PLACE:");
d1 = {a: 1};
d2 = {b: 2};
Object.assign(d1, d2); // in-place
console.log(d1);       // {a:1, b:2}

console.log();

console.log("SET NORMAL:");
let s1 = new Set([1, 2]);
let s2 = new Set([2, 3]);
let s3 = new Set([...s1, ...s2]); // novo
console.log(s1, s2, s3);

console.log("SET IN-PLACE:");
s1 = new Set([1, 2]);
s2 = new Set([2, 3]);
for (let x of s2) s1.add(x); // in-place
console.log(s1); // Set {1, 2, 3}

File: lab/python/tecnicas/dicionarios_e_sets_iplace.py
print("DICT NORMAL:")
d1 = {"a": 1}
d2 = {"b": 2}
d3 = {**d1, **d2}   # cria um novo
print(d1, d2, d3)   # {'a':1} {'b':2} {'a':1,'b':2}
print("DICT IN-PLACE")
d1 = {"a": 1}
d2 = {"b": 2}
d1.update(d2)
print(d1)           # {'a':1,'b':2} (alterado)

print("CONJUNTOS NORMAL:")
s1 = {1, 2}
s2 = {2, 3}
s3 = s1 | s2
print(s1, s2, s3)   # {1,2} {2,3} {1,2,3}
print("CONJUNTOS IN-PLACE:")
s1 = {1, 2}
s2 = {2, 3}
s1 |= s2
print(s1)           # {1,2,3}
